### PR TITLE
Added illusion Merchant

### PR DESCRIPTION
**Pull request by Haydrich**
> * **Addressed Issue(s)**:
> * **Server Mode**: Renewal
> * **Description of Pull Request**:

Added Illusion Merchant

- [x] NPC Shop Update Day
- [x] English Translation

### CODE REVIEW

File: npc/re/merchants/enchan_illusion_dungeons.txt [hunk 1]
```diff
@@ -1704,3 +1705,35 @@ OnInit:
...
+	switch( select( "Purchase with Zeny.", "Purchase with Illusion Stones" )) {
+		mes "[Illusion Merchant]";
+		mes "I hope you like it.";
+		close2;
+		case 1: callshop "market_resonance_stone"; end;
+		case 2: callshop "resonance_stone_barter"; end;
+	}
+OnInit:
+	.@npc_id = getnpcid(0);
+	setunitdata .@npc_id,UNPC_CLASS,2;
```
Comment (Atemo) [inline npc/re/merchants/enchan_illusion_dungeons.txt:1727]:
```suggestion
	setunitdata .@npc_id,UNPC_CLASS,4003;	// MAGICIAN_H
```

File: npc/re/merchants/enchan_illusion_dungeons.txt [hunk 2]
```diff
@@ -1704,3 +1705,35 @@ OnInit:
...
+	switch( select( "Purchase with Zeny.", "Purchase with Illusion Stones" )) {
+		mes "[Illusion Merchant]";
+		mes "I hope you like it.";
+		close2;
+		case 1: callshop "market_resonance_stone"; end;
+		case 2: callshop "resonance_stone_barter"; end;
+	}
+OnInit:
+	.@npc_id = getnpcid(0);
+	setunitdata .@npc_id,UNPC_CLASS,4003;	// MAGICIAN_H
```
Comment (Lemongrass3110) [inline npc/re/merchants/enchan_illusion_dungeons.txt:1727]:
```suggestion
	setunitdata .@npc_id,UNPC_CLASS, JOB_MAGE_HIGH;
```

File: npc/re/merchants/enchan_illusion_dungeons.txt [hunk 3]
```diff
@@ -1704,3 +1705,35 @@ OnInit:
...
+	setunitdata .@npc_id,UNPC_SEX,2;
```
Comment (Lemongrass3110) [inline npc/re/merchants/enchan_illusion_dungeons.txt:1728]: Please use the constant SEX_FEMALE (0) or SEX_MALE (1).

File: npc/re/merchants/enchan_illusion_dungeons.txt [hunk 4]
```diff
@@ -1704,3 +1705,35 @@ OnInit:
 	setunitdata .@npc_id, UNPC_HAIRCOLOR, 6;
 	end;
 }
+
+//============================================================
+//= Illusion Merchant
+//============================================================
+-	marketshop	market_resonance_stone	-1,100003:2000000:99999,100004:2000000:99999
+// prontera,88,113,5	script	Illusion Merchant#0829	MAGICIAN_H,{
```
Comment (Lemongrass3110) [inline npc/re/merchants/enchan_illusion_dungeons.txt:1713]:
```suggestion

```

File: npc/re/merchants/enchan_illusion_dungeons.txt [hunk 5]
```diff
@@ -1704,3 +1705,35 @@ OnInit:
...
+prontera,88,113,5	script	Illusion Merchant#0829	HIDDEN_NPC,{
+	mes "[Illusion Merchant]";
+	mes "I'm selling two random optional weapon grant scrolls that I made myself. You can choose the payment method you like.";
+	next;
+	switch( select( "Purchase with Zeny.", "Purchase with Illusion Stones" )) {
+		mes "[Illusion Merchant]";
```
Comment (Lemongrass3110) [inline npc/re/merchants/enchan_illusion_dungeons.txt:1719]: Its really strange that our script engine supports this, but please dont use this kind of syntax.

**Review by Lemongrass3110 — CHANGES_REQUESTED** [review]

**Comment by Lemongrass3110** [comment]
As for the shop update / market restock:
We have confirmed a few times already, that AEGIS does not persist the markets and this was a custom implementation by previous developers of rAthena or Hercules.
In reality AEGIS simply refills on each restart and because they have a fixed schedule for restarting it looks like a "refill" on a specific day.

**Review by Atemo — APPROVED** [review]

**Review by aleos89 — APPROVED** [review]

**Review by Atemo — APPROVED** [review]

File: npc/re/merchants/barters/enchan_illusion_dungeons.yml [hunk 6]
```diff
@@ -813,4 +827,4 @@ Body:
             Amount: 30
           - Index: 3
             Item: Token_Of_Bigstone
-            Amount: 150
+            Amount: 150
```
Comment (Atemo) [inline npc/re/merchants/barters/enchan_illusion_dungeons.yml:830]:
```suggestion
            Amount: 150

```

**Review by Atemo — APPROVED** [review]

File: npc/re/merchants/enchan_illusion_dungeons.txt [hunk 7]
```diff
@@ -846,3 +847,36 @@ OnInit:
...
+			end;
+	}
+OnInit:
+	.@npc_id = getnpcid(0);
+	setunitdata .@npc_id,UNPC_CLASS, JOB_MAGE_HIGH;
+	setunitdata .@npc_id,UNPC_SEX,SEX_FEMALE;
+	setunitdata .@npc_id,UNPC_HEADTOP,142;
+	setunitdata .@npc_id,UNPC_HEADMIDDLE,92;
+	setunitdata .@npc_id,UNPC_HAIRSTYLE,2;
+	setunitdata .@npc_id,UNPC_HAIRCOLOR,2;
```
Comment (Atemo) [inline npc/re/merchants/enchan_illusion_dungeons.txt:880]:
```suggestion
	setunitdata .@npc_id,UNPC_HAIRCOLOR,2;

	// Restock
	npcshopupdate "market_resonance_stone",100003,2000000,99999;
	npcshopupdate "market_resonance_stone",100004,2000000,99999;
```

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request merged**